go/vt/discovery/replicationlag.go

```diff
@@ -0,0 +1,72 @@
+package discovery
+
+import "fmt"
+
+// FilterByReplicationLag filters the list of EndPointStats by EndPointStats.Stats.SecondsBehindMaster.
+// The algorithm (EndPointStats that is non-serving or has error is ignored):
+// - Return the list if there is 0 or 1 endpoint.
+// - Return the list if all endpoints have <=30s lag.
+// - Filter by replication lag: for each endpoint, if the mean value without it is more than 0.7 of the mean value across all endpoints, it is valid.
+// For example, lags of (5s, 10s, 15s, 120s) return the first three;
+// lags of (30m, 35m, 40m, 45m) return all.
+func FilterByReplicationLag(epsList []*EndPointStats) []*EndPointStats {
+	list := make([]*EndPointStats, 0, len(epsList))
+	// filter non-serving endpoints
+	for _, eps := range epsList {
+		if !eps.Serving || eps.LastError != nil || eps.Stats == nil {
+			continue
+		}
+		list = append(list, eps)
+	}
+	if len(list) <= 1 {
+		return list
+	}
+	// if all have low replication lag (<=30s), return all endpoints.
+	allLowLag := true
+	for _, eps := range list {
+		if eps.Stats.SecondsBehindMaster > 30 {
```

Comment:
Done. SecondsBehindMaster is uint32 (not Duration), so I created a flag as Uint.